src/Map/Map.jsx

```diff
@@ -1,74 +1,247 @@
+      const {labels = [], positions = []} = line
+      const [start, end] = positions
+      const [startLabel, endLabel] = labels
+      const key = `${key}line${index}`
+      
+      return prev.concat([
+        <Polyline
+          color={line.color || color}
+          key={key}
+          positions={[start, end]}
+        >
+          {this.createPopupLabel(labels)}
+        </Polyline>,
+        this.createMarker({
+          color,
+          position: start,
+          labels: [startLabel]
+        }, `start${key}`),
+        this.createMarker({
+          color,
+          position: end,
+          labels: [endLabel]
+        }, `end${key}`)
+      ])
+    }, [])
+  }
+  
+  createPopupLabel (labels) {
+    const popupLabels = this.createLabels(labels)
+    
+    return popupLabels.length > 0
+      ? <Popup>
+          <div>
+            {popupLabels}
+          </div>
+        </Popup>
+      : null
+  }
+  
+  createLabels (labels) {
+    return labels.filter((label) => label)
+      .map((label, index) => (
+        <div
+          key={index}
+          style={{
+            fontWeight: index === 0 ? 'bold' : 'normal'
+          }}
+        >
+          {label}
+        </div>
+      ))
   }
 
   render () {
-    const {center, title} = this.props
+    const {center, mapOptions, tileLayerOptions, wms, zoomControlPosition} = this.props
+    let TileLayerComponent = wms ? WMSTileLayer : TileLayer
+    
     return (
       <LeafletMap
         center={center}
-        minZoom={3}
-        zoom={5}
-        zoomControl={true}
+        {...mapOptions}
       >
-        <LayersControl position='topright'>
-          <BaseLayer
-            checked
-            name='Color'
-          >
-            <TileLayer
-              attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
-              url='http://{s}.tile.osm.org/{z}/{x}/{y}.png'
-            />
-          </BaseLayer>
-          <BaseLayer name='Black & White'>
-            <TileLayer
-              attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
-              url='http://{s}.tiles.wmflabs.org/bw-mapnik/{z}/{x}/{y}.png'
-            />
-          </BaseLayer>
-          <Overlay
-            checked
-            name={title}
-          >
-            <LayerGroup>
-              {this.createMarkersList()}
-            </LayerGroup>
-          </Overlay>
+        <TileLayerComponent
+          attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+          url='http://{s}.tile.osm.org/{z}/{x}/{y}.png'
+          onLoad={this.loaded}
+          {...tileLayerOptions}
+        />
+        <LayerGroup>
+          {this.createBaseLayer()}
+        </LayerGroup>
+         <LayersControl position={zoomControlPosition}>
+          {this.createLayers()}
         </LayersControl>
       </LeafletMap>
     )
   }
-}
\ No newline at end of file
+}
+
+const allColors = ['blue',
```

Comment:
I would probably put this in a common location? Maybe we need to make a color scheme thats used in a bunch of stuff?

Also, the `'blue'` should be on the next line, not next to the `[`